File: 2419/Solution.java
class Solution {
    public int longestSubarray(int[] nums) {
        // TODO - Optimize this solution, only beats 7.5%
        int soln = 0;
        int cur_len = 0;
        int max_val = Arrays.stream(nums).max().getAsInt();
        for (int num : nums){
            if (num == max_val){
                cur_len = cur_len + 1;
            } else {
                soln = Math.max(cur_len, soln);
                cur_len = 0;
            }
        }
        return Math.max(cur_len, soln);
    }
}

File: 2419/solution.py
from typing import List


class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        soln = 0
        cur_len = 0
        max_val = max(nums)
        for num in nums:
            if max_val == num:
                cur_len += 1
            else:
                soln = max(cur_len, soln)
                cur_len = 0
        return max(cur_len, soln)
